ReactiveCocoa/Swift/Property.swift

```diff
@@ -95,6 +95,57 @@ extension MutableProperty: SinkType {
 	}
 }
 
+/// Wraps a `dynamic` property, or one defined in Objective-C, using Key-Value
+/// Coding and Key-Value Observing.
+///
+/// Use this class only as a last resort! `MutableProperty` is generally better
+/// unless KVC/KVO is required by the API you're using (for example,
+/// `NSOperation`).
+@objc public final class DynamicProperty: NSObject, PropertyType {
+	public typealias Value = AnyObject?
+
+	private weak var object: NSObject?
+	private let keyPath: String
+
+	/// The current value of the property, as read and written using Key-Value
+	/// Coding.
+	public var value: AnyObject? {
+		get {
+			return object?.valueForKeyPath(keyPath)
+		}
+
+		set(newValue) {
+			object?.setValue(newValue, forKeyPath: keyPath)
+		}
+	}
+
+	/// A producer that will create a Key-Value Observer for the given object,
+	/// send its initial value then all changes over time, and then complete
+	/// when the observed object has deallocated.
+	///
+	/// By definition, this only works if the object given to init() is
+	/// KVO-compliant. Most UI controls are not!
+	public var producer: SignalProducer<AnyObject?, NoError> {
+		if let object = object {
+			return object.rac_valuesForKeyPath(keyPath, observer: nil).asSignalProducer()
+				// Errors aren't possible, but the compiler doesn't know that.
+				|> catch { error in
+					fatalError("Received unexpected error from KVO signal: \(error)")
+					return .empty
```

Comment:
I guess it's because it fails to infer the return type, which you could do with `{ (() -> SignalProducer<(), NoError>), error in` but meh, not worth it.